### PR TITLE
New unit test: check for key existence

### PR DESCRIPTION
**Pull request by tisnik**
## Description

New unit test: check for key existence
Please note: not for WatsonX as it is handled differently

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [x] Unit tests

### CODE REVIEW

File: tests/unit/llms/test_llm_loader.py [hunk 1]
```diff
@@ -105,3 +105,22 @@ def test_constructor_when_missing_model_config():
...
+
+
+providers = [
+    constants.PROVIDER_OPENAI,
+    constants.PROVIDER_BAM,
+]
+
+
+@pytest.mark.parametrize("provider", providers)
+def test_constructor_no_keys(provider):
```
Comment (onmete) [inline tests/unit/llms/test_llm_loader.py:117]: Not sure what it is testing. That there is a raise on config when no API keys are configured?

Comment (tisnik) [reply tests/unit/llms/test_llm_loader.py:117]: yes. there's some `except->re raise` blocks we need to "touch"

**Pull request merged**